prov/shm/src/smr_msg.c

```diff
@@ -193,14 +193,16 @@ static ssize_t smr_generic_sendmsg(struct smr_ep *ep, const struct iovec *iov,
 	iface = smr_get_mr_hmem_iface(ep->util_ep.domain, desc, &device);
 
 	total_len = ofi_total_iov_len(iov, iov_count);
+	assert(!(op_flags & FI_INJECT) || total_len <= SMR_INJECT_SIZE);
```

Comment:
Would it be simpler to do:
```
if (op_flags & FI_INJECT) {
    assert(total_len <= SMR_INJECT_SIZE);
    goto sar;
}
```
with a sar label added below, so that we can skip checking op_flags & FI_INJECT repeatedly?